symengine/tests/cwrapper/test_cwrapper.c

```diff
@@ -238,6 +238,23 @@ void test_get_type() {
     basic_free(y);
 }
 
+void test_hash() {
+    basic x1, x2, y;
+    basic_init(x1);
+    basic_init(x2);
+    basic_init(y);
+    symbol_set(x1, "x");
+    symbol_set(x2, "x");
+    symbol_set(y, "y");
+
+    SYMENGINE_C_ASSERT(basic_hash(x1) == basic_hash(x2));
+    if (basic_hash(x1) != basic_hash(y)) SYMENGINE_C_ASSERT(basic_neq(x1,y));
```

Comment:
I got to know that even different objects have the same hash. But if they are different, they are indeed not equal. That is the way it is tested in `test_basic.cpp`.